### PR TITLE
fix(deps): update js-yaml to 5.4.1

### PR DESCRIPTION
**Pull request by mldangelo-oai**
## Summary

- Update `js-yaml` from 5.4.0 to 5.4.1 after #10629 landed, keeping the root, app, site, and lockfile versions aligned.
- Adopt upstream's merge-work accounting and bounded merge-sequence behavior for user-authored YAML.
- Add core and site regression tests for empty merge sources and oversized merge sequences.

## Validation

- Exact current-main branch: 279 focused core/manifest/config tests, 38 app YAML tests, and 8 site YAML tests passed.
- Root/app build, TypeScript, docs site build (`SKIP_OG_GENERATION=true`), lint, format check, and dependency consistency passed.
- No-cache CLI eval passed (1/1, score 1, zero errors); `npm audit` reported zero critical advisories (7 moderate, 28 high).
- The full local test suite was intentionally skipped; current-head CI is the full-matrix gate before merge.

### CODE REVIEW

**Comment by mldangelo-oai** [comment]
@codex review

File: site/src/utils/yaml.test.ts [hunk 1]
```diff
@@ -27,6 +27,17 @@ describe('loadYaml', () => {
     });
   });
 
+  it('limits empty merge sources and oversized merge sequences', () => {
```
[the remark this answers is not in the document]
Comment (mldangelo-oai) [reply site/src/utils/yaml.test.ts:30]: The site test is already in its own `docs(site): test js-yaml merge limits` commit (30c8de70a). The `fix(deps)` commit changes the dependency manifests, lockfile, and core test. The PR title follows the dependency-upgrade convention.
